client/dashboard/sites/site/environment-switcher.tsx

```diff
@@ -129,13 +142,41 @@ const EnvironmentSwitcher = ( { site }: { site: Site } ) => {
 		enabled: !! otherEnvironmentSiteId,
 	} );
 
+	const stagingSiteId = site.is_wpcom_staging_site ? site.ID : otherEnvironmentSiteId;
```

Comment:
Here we could consider using `getStagingSiteId` (from `/utils/site-staging-site`).